### PR TITLE
1820 add index to library page

### PR DESCRIPTION
**Pull request by solazio**
## Done
- _Add index navigation to the docstring libraries page._
- _Separate `docstring` and `source-code` in 2 views and routes._

## How to QA
- Or you can run the project locally using the [dotrun](https://snapcraft.io/dotrun) snap with `$ dotrun` and view it in your web browser at: http://localhost:8045/fran-wordpress/libraries/test
- And see it takes you the docstring page. Click on the `index` navigation at the top and see it works as expected.
- Click on the `Source code` button and see you are taken to http://localhost:8045/fran-wordpress/libraries/test/source-code and the page looks fine

## Issue / Card
Fixes https://github.com/canonical-web-and-design/snap-squad/issues/1820

## Screenshots
[if relevant, include a screenshot]

### CODE REVIEW

**Review by steverydz — APPROVED** [review]
LGTM 👍

File: templates/details/libraries/docstring.html [hunk 1]
```diff
@@ -4,18 +4,46 @@
...
+        <li class="p-list__item u-no-padding">
+          <ul class="p-list u-no-margin--bottom" style="margin-inline-start: 1.5rem;">
+            {% if not item["name"].startswith("__") %}
+            <li class="p-list__item">
+              <a href="#{{ group['name'].lower() }}-{{ item['name'].lower() }}">{% if item["type"].lower() == "function" %}def {{ item["name"] }}(){% else %}{{ item["type"].lower() }} {{ item["name"] }}{% endif %}</a>
+            </li>
+            {% endif %}
+          </ul>
+        </li>
+        {% endfor %}
```
Comment (jkfran) [inline templates/details/libraries/docstring.html:36]: It would be more readable with more indentation here

**Review by jkfran — APPROVED** [review]
Looks good to me

File: templates/details/libraries/docstring.html (hunk 1 above)
Comment (solazio) [reply templates/details/libraries/docstring.html:36]: agree, will fix it

**Pull request merged**